chore!: remove library v3

kikoso opened the pull request:
This PR removes the V3 of the library.

---

Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [X] Make sure to open a GitHub issue as a bug/feature request before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [X] Ensure the tests and linter pass
- [X] Code coverage does not decrease (if any source code was changed)
- [X] Appropriate docs were updated (if necessary)
- [X] Will this cause breaking changes to existing Java or Kotlin integrations? If so, ensure the commit has a `BREAKING CHANGE` footer so when this change is integrated a major version update is triggered. See: https://www.conventionalcommits.org/en/v1.0.0/

wangela commented:
Added ! to the PR title to indicate breaking change, but will need to manually add this to the squashed commit message.

kikoso commented on README.md:
```diff
@@ -32,7 +32,7 @@ You can view the generated [reference docs][javadoc] for a full list of classes
 
 * Android API level 15+
-* Maps SDK via Google Play Services ~OR (Deprecated) [Maps SDK v3 BETA] library~
+* Maps SDK via Google Play Services OR [Maps SDK] library
```
> ```
> * [Maps SDK for Android](https://developers.google.com/maps/documentation/android-sdk/config) via Google Play Services
> ```

Changed

kikoso replied:
> Please also delete L44-45 and other mentions of v3 in the README under ## Demo App

I have left only the migration guide, which might make sense for users with the legacy artifact.

kikoso commented on demo/build.gradle:
```diff
@@ -68,55 +54,21 @@ dependencies {
     // GMS
-    gmsImplementation 'com.google.android.gms:play-services-maps:17.0.1'
-    gmsImplementation project(':library')
+    implementation 'com.google.android.gms:play-services-maps:17.0.1'
```
> Will changing this cause a breaking change for developers?

Changing the Gradle implementation command? This is the demo app, users integrating the library are likely doing it via Maven and updating the version number (see the latest dependabot PR in the `android-samples` [here](https://github.com/googlemaps/android-samples/pull/1146#pullrequestreview-1240451474))

kikoso commented on demo/build.gradle:
```diff
@@ -68,55 +54,21 @@ dependencies {
-    v3Implementation 'com.android.volley:volley:1.2.1' // TODO - Remove this after Maps SDK v3 beta includes Volley versions on Maven Central
-    v3Implementation project(':library-v3')
-    v3Implementation 'androidx.multidex:multidex:2.0.1'
-    implementation "androidx.lifecycle:lifecycle-viewmodel-ktx:2.5.1"
```
L80 is included in the library, and L79 is a sub-dependency from the Play Services Map and the Compatibility package.

On a second thought, it will be a better practice to lock on these versions, instead of relying on the version other libraries use to include them. I have re-added them.

kikoso commented:
> Added ! to the PR title to indicate breaking change, but will need to manually add this to the squashed commit message.

I have rebased from `main` and squashed all the commits into a single one with the exclamation symbol, let me know if this works out.
